drivers/tty/serdev/pibridge.c

```diff
@@ -87,16 +87,18 @@ static int pibridge_discard_timeout(u16 len, u16 timeout)
 	struct pibridge *pi = pibridge_s;
 	int ret;
 
-	wait_event_interruptible_timeout(pi->read_queue,
-		kfifo_len(&pi->read_fifo) >= len, msecs_to_jiffies(timeout));
-
-	mutex_lock(&pi->lock);
-	if (kfifo_len(&pi->read_fifo) >= len)
-		ret = 0;
-	else
-		ret = -1;
-	kfifo_reset(&pi->read_fifo);
-	mutex_unlock(&pi->lock);
+	ret = wait_event_interruptible_timeout(pi->read_queue,
+					       kfifo_len(&pi->read_fifo) >= len,
+					       msecs_to_jiffies(timeout));
+	if (ret != -ERESTARTSYS) {
```

Comment:
Why not

```if (ret == -ERESTARTSYS)
	goto out;
```

and insert an `out:` label before the `mutex_unlock()`? Avoids the need to indent all those lines. At least that's the pattern I've seen used most of the time in the kernel.